nmostesting/suites/IS1101Test.py

```diff
@@ -0,0 +1,3813 @@
+                    ):
+                        inputs.append(input_id)
+                    else:
+                        print(
+                            "Inputs {} are not connected or does'nt support base Edid".format(
+                                input_id
+                            )
+                        )
+                        break
+            except json.JSONDecodeError:
+                return test.FAIL("Non-JSON response returned from Node API")
+            except KeyError as e:
+                return test.FAIL("Unable to find expected key: {}".format(e))
+
+            if len(inputs) == 0:
+                return test.UNCLEAR("No input supports changing the base EDID")
+            for input_id in inputs:
+                valid, response = TestHelper.do_request(
+                    "GET", self.compat_url + "inputs/" + input_id + "/properties/"
+                )
+                if not valid:
+                    return test.FAIL(
+                        "Unexpected response from the streamcompatibility API: {}".format(
+                            response
+                        )
+                    )
+                if response.status_code != 200:
+                    return test.FAIL(
+                        "The input {} properties streamcompatibility request has failed: {}".format(
+                            input_id, response
+                        )
+                    )
+                try:
+                    version = response.json()["version"]
+                except json.JSONDecodeError:
+                    return test.FAIL("Non-JSON response returned from Node API")
+                except KeyError as e:
+                    return test.FAIL("Unable to find expected key: {}".format(e))
+
+                self.version[input_id] = version
+
+                valid, response = TestHelper.do_request(
+                    "GET", self.node_url + "senders/" + sender_id
+                )
+                if not valid:
+                    return test.FAIL(
+                        "Unexpected response from the Node API: {}".format(response)
+                    )
+                if response.status_code != 200:
+                    return test.FAIL(
+                        "The sender {} is not available in the Node API request: {}".format(
+                            sender_id, response.json()
+                        )
+                    )
+                try:
+                    version = response.json()["version"]
+                except json.JSONDecodeError:
+                    return test.FAIL("Non-JSON response returned from Node API")
+                except KeyError as e:
+                    return test.FAIL("Unable to find expected key: {}".format(e))
+
+                self.version[sender_id] = version
+
+                default_edid = self.get_effective_edid(test, input_id)
+
+                self.another_grain_rate_constraints[sender_id] = {
```

Comment:
`another_grain_rate_constraints` only appears to be used locally within this test. Why is it a class variable?